internal/operators/ocs/ocs_operator.go

```diff
@@ -93,23 +93,26 @@ func (o *operator) ValidateHost(_ context.Context, cluster *common.Cluster, host
 		return api.ValidationResult{Status: api.Failure, ValidationId: o.GetHostValidationID(), Reasons: []string{err.Error()}}, nil
 	}
 
+	role := common.GetEffectiveRole(host)
+
+	// auto-assign effective role (meaning both role and suggested-role) should be a transient
+	// state. return pending status in this case
+	if role == models.HostRoleAutoAssign {
+		status := "Host role must be assigned or evaluated to master or worker."
+		return api.ValidationResult{Status: api.Pending, ValidationId: o.GetHostValidationID(), Reasons: []string{status}}, nil
+	}
+
```

Comment:
So basically leave I should leave the logic as it was before with the distinction between compact mode and standard mode right?

Reply:
Yes, either that or if you are changing it here to show validation pending in case of auto-assign then even in ValidateCluster in case of auto-assign role(both in mode), we should show validation pending